rules/rules-reviewed/camel3/camel2/xml-315-changes.windup.xml

```diff
@@ -0,0 +1,236 @@
+        </dependencies>
+        <sourceTechnology id="camel" versionRange="[3.0,3.15)"/>
+        <targetTechnology id="camel" versionRange="[3.15,)" />
+    </metadata>
+    <rules>
+        <rule id="xml-315-00001">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="org.apache.camel" artifactId="{*}"/>
+                    </project>
+                    <dependency groupId="org.apache.camel" artifactId="{*}"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="JDK 8 no longer supported" effort="7" category-id="mandatory" >
+                    <message>JDK 8 no longer supported - In Camel 3.15, JDK 11 or 17 is required.</message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_15.html#_upgrading_camel_3_14_to_3_15" title="Camel 3 - Migration Guide: JDK Upgrade" />
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="xml-removed-camel315-00001">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="org.apache.camel" artifactId="camel-kamelet-reify" />
+                    </project>
+                    <dependency groupId="org.apache.camel" artifactId="camel-kamelet-reify" />
+                </or>
+            </when>
+            <perform>
+                <hint title="org.apache.camel:camel-kamelet-reify component has been retired" effort="7" category-id="mandatory" >
+                    <message>`camel-kamelet-reify` component has been retired in Apache Camel 3.15.  </message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_15.html#_removed_components" title="Camel 3 - Migration Guide: Removed components" />
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="xml-removed-camel315-00002">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="org.apache.camel" artifactId="camel-spring-javaconfig" />
+                    </project>
+                    <dependency groupId="org.apache.camel" artifactId="camel-spring-javaconfig" />
+                </or>
+            </when>
+            <perform>
+                <hint title="org.apache.camel:camel-spring-javaconfig component has been retired" effort="7" category-id="mandatory" >
+                    <message>`camel-spring-javaconfig` component has been retired in Apache Camel 3.15.</message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_15.html#_removed_components" title="Camel 3 - Migration Guide: Removed components" />
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="properties-removed-camel315-00001">
+            <when>
+                <filecontent pattern="camel.dataformat.fhirjson.{*}={*}" filename="{*}.properties"/>
```

Comment:
I think these patterns could be simplified to `camel.dataformat.fhirjson.{*}` given that we are looking into `properties` files. What do you think? It would also apply to the rest. This would enhance performance a bit, since `filecontent` rules tend to be a bit slower.

Reply:
On property key name changes, I really want to match to as close to the setting as possible, and my thought was that we shouldn't match in a case where there isn't a setting - if the key is mentioned in a context without the =, don't match.